tests/unit/test_plugin.py

```diff
@@ -58,7 +58,49 @@ def test_get_account_info(mock_urlopen):
     }
 
 
-def test_meter_billing():   # Currently no-op
+@patch('csp_billing_adapter_google.plugin.urllib.request.urlopen')
+def test_fetch_metadata_fail(mock_urlopen):
+    urlopen = MagicMock()
+    urlopen.read.side_effect = [
```

Comment:
Minor code pattern inconsistency between here and line 95, test_meter_billing_fail(), below, where here you have a one element list containing the exception, for the side effect value, where as below you have side_effect being assigned the exception... I would go with the latter code pattern here also since, for your test scenario, you want it to always fail with the exception whenever invoked...

Reply:
I hope it is OK now and I have removed WIP.